src/dmd/dcast.d

```diff
@@ -1094,6 +1094,19 @@ MATCH implicitConvTo(Expression e, Type t)
             visit(cast(Expression)e);
         }
 
+        override void visit(AndExp e)
```

Comment:
@WalterBright - This override is identical to OrExp and XorExp.  I've checked a couple other places (expressionsem, optimize) and it's all the same structure as well except for the specific operator.

Maybe these three classes could be merged into a BitwiseExp node similar to the LogicalExp refactoring.